src/Test/Utilities/Portable/Compilation/ControlFlowGraphVerifier.cs

```diff
@@ -671,6 +675,49 @@ string getRegionId(ControlFlowRegion region)
             {
                 return $"R{regionMap[region]}{idSuffix}";
             }
+
+            string getOperationTree(IOperation operation)
+            {
+                var walker = new OperationTreeSerializer(graph, currentRegion, idSuffix, anonymousFunctionsMap, compilation, operation, initialIndent: 8 + indent);
+                walker.Visit(operation);
+                return walker.Builder.ToString();
+            }
+        }
+
+        private class OperationTreeSerializer : OperationTreeVerifier
+        {
+            private readonly ControlFlowGraph _graph;
+            private readonly ControlFlowRegion _region;
+            private readonly string _idSuffix;
+            private readonly Dictionary<IFlowAnonymousFunctionOperation, ControlFlowGraph> _anonymousFunctionsMap;
+
+            public OperationTreeSerializer(ControlFlowGraph graph, ControlFlowRegion region, string idSuffix, 
+                                           Dictionary<IFlowAnonymousFunctionOperation, ControlFlowGraph> anonymousFunctionsMap,
+                                           Compilation compilation, IOperation root, int initialIndent) :
+                base(compilation, root, initialIndent)
+            {
+                _graph = graph;
+                _region = region;
+                _idSuffix = idSuffix;
+                _anonymousFunctionsMap = anonymousFunctionsMap;
+            }
+
+            public System.Text.StringBuilder Builder => _builder;
+
+            public override void VisitFlowAnonymousFunction(IFlowAnonymousFunctionOperation operation)
+            {
+                base.VisitFlowAnonymousFunction(operation);
+
+                LogString("{");
+                LogNewLine();
+                var g = _graph.GetAnonymousFunctionControlFlowGraph(operation);
+                int id = _anonymousFunctionsMap.Count;
```

Comment:
>_anonymousFunctionsMap [](start = 25, length = 22)

Does anyone care about the contents of `_anonymousFunctionsMap`? Or do we just need to keep track of the number of anonymous functions encountered? Consider using/incrementing an `int` if that is sufficient.

Reply:
> Does anyone care about the contents of _anonymousFunctionsMap? Or do we just need to keep track of the number of anonymous functions encountered? Consider using/incrementing an int if that is sufficient.

line 309

---
In reply to: [193840322](https://github.com/dotnet/roslyn/pull/27536#discussion_r193840322) [](ancestors = 193840322)

Reply:
Thanks.
It may be worth a comment on the class to say this serializer prints out anonymous functions, and accumulates them in the passed-in `anonymousFunctionsMap`

---
In reply to: [193841284](https://github.com/dotnet/roslyn/pull/27536#discussion_r193841284) [](ancestors = 193841284,193840322)